### PR TITLE
azure: Fix some minor issues which have broken our configuration 

### PR DESCRIPTION
**Pull request by alexcrichton**
* Ensure that when we enable IPv6 for Docker on Linux that the various directories before writing a config file
* Delete a previously installed rustup if any since it seems to interfere with Cargo's test suite.

### CODE REVIEW

File: .azure-pipelines/steps/run.yml [hunk 1]
```diff
@@ -101,6 +102,7 @@ steps:
 
 - bash: |
     set -e
+    rustup self uninstall -y || true
```
Comment (pietroalbini) [inline .azure-pipelines/steps/run.yml]: Can you add the reason why rustup is being uninstalled in a comment? r=me otherwise!

**Comment by pietroalbini** [comment]
@alexcrichton wait, you reapproved this after it was included in a rollup currently testing.

**Comment by alexcrichton** [comment]
Ah that was my bad, I pushed a commit here by accident and then removed the commit, but that messes up bors. In any case looks like it got figured out.

**Pull request merged**